### PR TITLE
added async tracking info to the readme

### PR DESCRIPTION
**Pull request by bartes**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -100,3 +100,26 @@ end
 ## Signature
 
 `Castle::SecureMode.signature(user_id)` will create a signed user_id.
+
+## Async tracking
+
+By default Castle sends requests synchronously. To eg. use Sidekiq to send requests in a background worker you can pass data to the worker:
```
Comment (nijikon) [inline README.md]: By default Castle sends tracking requests synchronously. To use Sidekiq to send requests to a background worker you can pass data to the worker:

File: README.md [hunk 2]
```diff
@@ -100,3 +100,26 @@ end
...
+
+#### castle_tracking_worker.rb
+
+```
+class CastleTrackingWorker
+  include Sidekiq::Worker
+
+  def perform(request, context = nil, track_options = {})
+    client = ::Castle::Client.new(request, {context: context})
```
Comment (nijikon) [inline README.md]: `client = ::Castle::Client.new(request, { context: context })`

File: README.md [hunk 3]
```diff
@@ -100,3 +100,26 @@ end
...
+    client.track(track_options)
+  end
+end
+```
+
+#### tracking_controller.rb
+
+```
+CastleTrackingWorker.perform_async(request, optional_context, {})
```
Comment (nijikon) [inline README.md]: `CastleTrackingWorker.perform_async(request)`

Comment (bartes) [reply README.md]: I want to show where and how to pass additional context data

File: README.md (hunk 1 above)
Comment (bartes) [reply README.md]: should we use 'tracking' word here ? Fine with removing 'eg'. Why `in` and not `to` ?

**Pull request merged**